src/main/java/ch/njol/skript/expressions/ExprAllBannedPlayers.java

```diff
@@ -0,0 +1,76 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("All Banned Players")
+@Description("Returns a list of all the banned players.")
+@Examples({
+	"command /banlist:",
+	"\ttrigger:",
+	"\t\tsend all the banned players"
+})
+@Since("INSERT VERSION")
+public class ExprAllBannedPlayers extends SimpleExpression<OfflinePlayer> {
+
+	static {
+		Skript.registerExpression(ExprAllBannedPlayers.class, OfflinePlayer.class, ExpressionType.SIMPLE, "[(all [[of] the]|the)] banned players");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected OfflinePlayer[] get(Event event) {
+		return Bukkit.getBannedPlayers().toArray(new OfflinePlayer[0]);
```

Comment:
Do you know if this accesses the file? If so, will this block?

Reply:
Bukkit#getBannedPlayers() returns net.minecraft.server.players.PlayerList#bans, which loads all bans from banned-player.json on instantiation of PlayerList.

It is not blocking.